app/assets/javascripts/package.json

```diff
@@ -22,9 +22,9 @@
   "resolutions": {
     "**/babel-plugin-debug-macros": "npm:@discourse/babel-plugin-debug-macros@0.4.0-pre1"
   },
-  "devDependencies": {},
-  "dependencies": {
+  "devDependencies": {
     "patch-package": "^6.5.1",
```

Comment:
only moved these deps to devDeps, because... it makes more sense for them to be devDeps

Reply:
We currently don't install dev deps in most of production deployments so `patch-package` has to stay in `dependencies` for now 😃

Reply:
most of the JS ecosystem doesn't follow this pattern :sweat_smile: :see_no_evil:

Reply:
> would it be worth adding a lint

It would probably be better for us to put that effort into making our setup less surprising. It shouldn't be too hard for us to change strategies.

> most of the JS ecosystem doesn't follow this pattern 😅 🙈

So the standard everywhere is to put production dependencies into `devDependencies`? 🤔 How would you decide what goes into `devDependencies` vs `dependencies`?

Reply:
> the app, as authored, doesn't actually run in production

Ah I see, so `devDependencies` is essentially "build time dependencies" - makes sense. Thanks for the detailed explanation ❤️ 

I opened a PR over in discourse_docker to get the ball rolling on improving this: https://github.com/discourse/discourse_docker/pull/708

But anyway, as far as this virtual-dom PR is concerned, I think we're good to go - thanks!